clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -290,8 +291,10 @@ func GetReplicasFromRuntimeObject(c clientset.Interface, obj runtime.Object) (in
 		}
 		return 0, nil
 	case *appsv1.DaemonSet:
-		// TODO(#790): In addition to nodeSelector the affinity should be also taken into account
-		return getNumSchedulableNodesMatchingSelector(c, typed.Spec.Template.Spec.NodeSelector)
+		if typed.Spec.Template.Spec.Affinity != nil && typed.Spec.Template.Spec.Affinity.NodeAffinity != nil {
```

Comment:
nit - I would prefer passing types.Spec.Template.Spec.Affinity here and do the nil handling in the getNumSchedu... function